src/shutdown_manager.ts

```diff
@@ -31,9 +35,25 @@ export default class ShutdownManager {
         const key = this.shutDownKey(details.instanceId);
         const res = await this.redisClient.get(key);
         ctx.logger.debug('Read shutdown status', { key, res });
-        if (res == 'shutdown') {
-            return true;
-        }
-        return false;
+        return res == 'shutdown';
+    }
+
+    async setScaleDownProtected(
+        ctx: Context,
+        details: InstanceDetails,
```

Comment:
Do we need the details here?  We seem to only use the instanceId, so perhaps it makes more sense to only pass the instanceId?